src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionProcessor.java

```diff
@@ -500,8 +501,8 @@ private void distinguishPointsFromShapePoints(final WaySectionChangeSet changeSe
         final Time time = Time.now();
         logger.info("Started Shape Point Detection for Shard {}", getShardOrAtlasName());
 
-        Iterables.stream(this.rawAtlas.points()).filter(point -> isAtlasPoint(changeSet, point))
-                .forEach(changeSet::recordPoint);
+        StreamSupport.stream(this.rawAtlas.points().spliterator(), true)
```

Comment:
Out of genuine curiosity, how much performance gain do you get by using spliterator here?

Reply:
~3x increase in performance.